vernac/metasyntax.ml

```diff
@@ -37,41 +37,90 @@ let pr_entry e =
   let () = Pcoq.Entry.print ft e in
   str (Buffer.contents entry_buf)
 
+let error_unknown_entry ?loc name =
+  user_err ?loc Pp.(str "Unknown or unprintable grammar entry " ++ str name ++ str".")
+
 let pr_registered_grammar name =
   let gram = Pcoq.find_grammars_by_name name in
   match gram with
-  | [] -> user_err Pp.(str "Unknown or unprintable grammar entry.")
+  | [] -> error_unknown_entry name
   | entries ->
-    let pr_one (Pcoq.AnyEntry e) =
+    let pr_one (Pcoq.Entry.Any e) =
       str "Entry " ++ str (Pcoq.Entry.name e) ++ str " is" ++ fnl () ++
       pr_entry e
     in
     prlist pr_one entries
 
-let pr_grammar = function
+let pr_grammar_subset grammar =
+  let pp = String.Map.mapi (fun name l -> match l with
+      | []  -> assert false
+      | entries ->
+        str "Entry " ++ str name ++ str " is" ++ fnl() ++
+        prlist_with_sep (fun () -> str "or" ++ fnl())
+          (fun (Pcoq.Entry.Any e) -> pr_entry e)
+          entries)
+      grammar
+  in
+  let pp = CString.Map.bindings pp in
+  prlist_with_sep fnl (fun (_,pp) -> pp) pp
+
+let is_known = let open Pcoq.Entry in function
   | "constr" | "term" | "binder_constr" ->
```

Comment:
Handling these "known" special names (constr,vernac,whatever is given to Pcoq.register_grammars_by_name) is not as useful now but for now is kept.